coding/python_style_guide.rst

```diff
@@ -467,40 +474,42 @@ This comment should appear after the ``def`` line.
 =====================
 
 We follow `PEP 8ʼs naming conventions <https://www.python.org/dev/peps/pep-0008/#naming-conventions>`_, with exceptions listed here.
-The naming conventions for LSST Python and C++ source have been defined to be as similar as the respective languages allow.
+C++ source code included within a Python package would normally follow the naming conventions of the Python package for APIs that are to be visible to Python users.
 
-In general:
+All LSST Python source code is consistent with :pep:`8` naming in the following ways:
 
 - class names are ``CamelCase`` with leading uppercase,
 - module variables used as module global constants are ``UPPERCASE_WITH_UNDERSCORES``,
-- all other names are ``camelCase`` with leading lowercase.
+
+Some packages, for historical reasons, do not fully adhere to :pep:`8`.
+These packages, and the associated naming conventions, are described in :ref:`style-guide-py-sci-pi-naming`.
+Code in other packages SHOULD be consistent within a top-level package built by Jenkins, or within a distinct service, and it is RECOMMENDED that :pep:`8` naming convention be adopted for these, whilst understanding that it may be difficult to modify existing packages.
+Consistency within a package is important.
+Within these stated constraints new packages SHOULD use :pep:`8` naming conventions.
 
 Names may be decorated with leading and/or trailing underscores.
 
-.. _style-guide-py-2-2:
+.. _style-guide-py-sci-pi-naming:
 
-User defined names SHOULD NOT shadow python built-in functions
---------------------------------------------------------------
+Naming Conventions for Science Pipelines
+----------------------------------------
 
-Names which shadow a python built-in function may cause confusion for readers of the code.
-Creating a more specific identifier is suggested to avoid collisions.
-In the case of *filter*, ``filterName`` may be appropriate; for *filter objects*, something like ``filterObj`` might be appropriate.
+For historical reasons, Science Pipelines code (nominally, all packages included in the ``lsst_apps`` metapackage, as well as ``meas_*``, ``pipe_*``, and ``obs_*`` and all dependencies), uses ``camelCase`` rather than completely adhering to :pep:`8`-style.
```

Comment:
`camelCase` is used only sometimes.  I'd change "uses camelCase rather than completely adhering" to "does not completely adhere".